src/nwb_conversion_tools/datainterfaces/ophys/baseimagingextractorinterface.py

```diff
@@ -77,10 +77,18 @@ def run_conversion(
         nwbfile: Optional[NWBFile] = None,
         metadata: Optional[dict] = None,
         overwrite: bool = False,
+        stub_test: bool = False,
         save_path: OptionalFilePathType = None,
     ):
+
+        if stub_test:
+            stub_frames = min([100, self.imaging_extractor.get_num_frames()])
```

Comment:
I was thinking we should set this to

```python
stub_frames = min([max(self.imaging_extractor.get_image_size()) + 1, self.imaging_extractor.get_num_frames()])
```

to avoid triggering the `check_data_orientation` from the NWBInspector.

Reply:
I feel that we should not make these type of decisions base on the failure in a edge case of the inspector. 

I actually suggest two things to avoid the problem with our own stubbed files:
1) If available, used the property `dimension` in the [TwoPhotonSeries](https://pynwb.readthedocs.io/en/stable/pynwb.ophys.html#pynwb.ophys.TwoPhotonSeries) to figure out which axes are the ones corresponding to the images (we can actually set this by default in our conversion pipeline using `imaging_extractor.get_image_size()`)
2) When the property `dimension` is not available, `nwbinspector` could check for two equal size (or close enough) sizes for the images (usually images are equal rows and columns).

These solution has the advantage that will generalize beyond our use case here.

Reply:
> I feel that we should not make these type of decisions base on the failure in a edge case of the inspector.

It's not an edge case, the NWBInspector correctly raises in the situation, it's just that the real data doesn't match this situation.

It would be nice if the NWBInspector reports accurately reflected what we would see if run on the full file as opposed to a stub, since the purpose of a stub is to have a 'smaller' version of the file that looks as close as possible to the full one.

We can of course always mentally ignore it in the report, but still, I don't see what the big deal is about making the report as reflective as possible.

> 1. If available, used the property dimension in the [TwoPhotonSeries](https://pynwb.readthedocs.io/en/stable/pynwb.ophys.html#pynwb.ophys.TwoPhotonSeries) to figure out which axes are the ones corresponding to the images (we can actually set this by default in our conversion pipeline using imaging_extractor.get_image_size())
> 2. When the property dimension is not available, nwbinspector could check for two equal size (or close enough) sizes for the images (usually images are equal rows and columns).

This is an extremely generic check on all time series and it works simply by asserting that the longest axis is the zero axis (time). For small images (no axis above 100 in pixels) this is already guaranteed by the hardcoded value, but would then become triggered for large images simply because the value is hardcoded.

Other checks do more sophisticated things for particular objects, similar to what you are suggesting. But those are other checks, not this check.

Reply:
Just an idea - we can always add in later (together with other interfaces, as well)